packages/amplify-category-analytics/provider-utils/awscloudformation/service-walkthroughs/kinesis-walkthrough.js

```diff
@@ -0,0 +1,208 @@
+const path = require('path');
+// FIXME: may be removed from here, since addResource can pass category to addWalkthrough
+const category = 'analytics';
+const service = 'Kinesis';
+
+async function addWalkthrough(context, defaultValuesFilename, serviceMetadata) {
+  const resourceName = resourceAlreadyExists(context);
+
+  if (resourceName) {
+    context.print.warning('Kinesis resource have already been added to your project.');
+    process.exit(0);
+  }
+  return configure(context, defaultValuesFilename, serviceMetadata);
+}
+
+function migrate() {
+  // no-op for now
+}
+
+function configure(context, defaultValuesFilename, serviceMetadata, resourceName = null) {
+  const { amplify } = context;
+  const { inputs } = serviceMetadata;
+  const defaultValuesSrc = `${__dirname}/../default-values/${defaultValuesFilename}`;
+  const { getAllDefaults } = require(defaultValuesSrc);
+  const defaultValues = getAllDefaults(amplify.getProjectDetails());
+  const projectBackendDirPath = amplify.pathManager.getBackendDirPath();
+
+  const questions = inputs
+    .map(input => ({
+      name: input.key,
+      message: input.question,
+      type: input.type || 'input',
+      choices: input.options || undefined,
+      required: input.required || false,
+      validate: 'validation' in input ? amplify.inputValidation(input) : undefined,
+      default: () => {
+        const defaultValue = defaultValues[input.key];
+        return defaultValue;
+      },
+    }))
+    // when resourceName is provider, we are in update flow - skip name question
+    .filter(question => (resourceName && question.name !== 'kinesisStreamName') || !resourceName);
+
+  return inquirer.prompt(questions).then(async answers => {
+    const targetResourceName = resourceName || answers.kinesisStreamName;
+    const shardCount = answers.kinesisStreamShardCount;
+    const templateDir = `${__dirname}/../cloudformation-templates`;
+    const resourceDirPath = path.join(projectBackendDirPath, category, targetResourceName);
+
+    if (!resourceName && resourceNameAlreadyExists(context, targetResourceName)) {
+      throw new Error(`Resource ${targetResourceName} already exists in ${category} category.`);
+    }
+
+    const copyJobs = [
+      {
+        dir: templateDir,
+        template: serviceMetadata.cfnFilename,
+        target: path.join(resourceDirPath, serviceMetadata.cfnFilename),
+        paramsFile: path.join(resourceDirPath, 'parameters.json'),
+      },
+    ];
+
+    const params = {
+      kinesisStreamName: targetResourceName,
+      kinesisStreamShardCount: shardCount,
+      authRoleName: defaultValues.authRoleName,
+      unauthRoleName: defaultValues.unauthRoleName,
+      authPolicyName: defaultValues.authPolicyName,
+      unauthPolicyName: defaultValues.unauthPolicyName,
+    };
+
+    // Check for authorization rules and settings
+
+    const { checkRequirements, externalAuthEnable } = require('amplify-category-auth');
+
+    const apiRequirements = {
```

Comment:
why is it called apiRequirements when it is kinesis? copy and paste or intentional? also we're getting 'api' requirements below.

Reply:
`copy and paste` from [kinesis-walkthrough.js#L75](https://github.com/ambientlight/amplify-cli/blob/b0af83b299dd9786b9c48ffdc5c86e37f33377fc/packages/amplify-category-analytics/provider-utils/awscloudformation/service-walkthroughs/kinesis-walkthrough.js#L75)

looks legit to me, am I missing something here?